db/logs.go

```diff
@@ -63,3 +63,14 @@ func (d *LogsDB) Logs(ctx context.Context, dealID uuid.UUID) ([]DealLog, error)
 
 	return dealLogs, nil
 }
+
+func (d *LogsDB) CleanupLogs(ctx context.Context, duration int) error {
+
+	t := time.Now()
+	td := t.AddDate(0, 0, -1*duration)
+
+	qry := "DELETE from DealLogs WHERE DealUUID IN (SELECT DealUUID FROM DealLogs WHERE CreatedAt < ?)"
```

Comment:
Most databases have an upper limit on the number of values that can be in an IN clause. So just in case I'd suggest adding DISTINCT to the sub-select:
```suggestion
	qry := "DELETE from DealLogs WHERE DealUUID IN (SELECT DISTINCT DealUUID FROM DealLogs WHERE CreatedAt < ?)"
```